src/Disciplina/Launcher/Resource.hs

```diff
@@ -1,27 +1,70 @@
+{-# LANGUAGE FunctionalDependencies #-}
 
 -- | Common resources used by Disciplina nodes
 
 module Disciplina.Launcher.Resource
        ( BasicNodeResources (..)
-       , bracketBasicNodeResources
+       , BracketResource(..)
        ) where
 
 import Universum
 
-import Mockable (Production)
-import System.Wlog (LoggerConfig (..), LoggerName, {-WithLogger,-} maybeLogsDirB, parseLoggerConfig,
-                    productionB, removeAllHandlers, setupLogging, showTidB)
+import Control.Monad.Cont (ContT (..))
+import System.Wlog (LoggerConfig (..), LoggerName, maybeLogsDirB, parseLoggerConfig, productionB,
+                    removeAllHandlers, setupLogging, showTidB)
 
-import Disciplina.DB.Real (NodeDB, closeNodeDB, openNodeDB)
+import Disciplina.DB.Real (DBParams, NodeDB, bracketNodeDB)
 import Disciplina.Launcher.Params (BasicNodeParams (..), LoggingParams (..))
 
 -- | Datatype which contains resources required by all Disciplina nodes
 -- to start working.
 data BasicNodeResources = BasicNodeResources
     { bnrLoggerName :: !LoggerName
-    , bnrDB         :: !NodeDB
     }
 
+----------------------------------------------------------------------------
+-- Resources
+----------------------------------------------------------------------------
+
+{- No top-level @acquire@s and @release@s for intension, having bracket built
+from brackets of smaller resources automatially forces proper coupling of
+acquire and release operations and so it's easier to use in various types of
+node which require different resources.
+-}
+
+-- | Resources construction.
+class BracketResource param res | param -> res, res -> param where
```

Comment:
1. This will not work for the same reasons you can't use global `ResourceT` if you intend to fork later (and we definitely will).
2. In any case I would recommend to use standard resource allocation frameworks, as already mentioned `resourcet` or `componentm`.

Reply:
Could you elaborate on "global `ResourceT`"? Some scenario when we fork threads and don't wait for them to complete?

Reply:
@Martoon-00 
Misha is right, I missed the fact that `StM (ContT m a)` isn't equal to `m a`, which means that forking inside it is unsafe. `ResourceT` is basically very similar to your construction with `ContT`, and we already had problems with `ResourseT` and concurrency in Cardano because of that - every forked process loses the exception handling continuation. 
We have two options: don't try to use pretty abstractions for resource handling at all and do `release` and `acquire` manually, or think very carefully about how can we use `ResourceT` (or similar) with `Control.Concurrent.Async.Lifted` (without `Safe`).